instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java

```diff
@@ -57,14 +57,19 @@ public static void nameResourceAndStartSpan(
         @Advice.Argument(2) Object handler,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
+      // TODO (trask) should there be a way to customize Instrumenter.shouldStart()?
```

Comment:
Is this the usual nested span problem and not specific to instrumenter API?

Reply:
interesting, ya, we could look at it that way, if we introduce a `HandlerSpan` (similar to `ServerSpan` and `ClientSpan`), i'll open an issue to discuss